collectd/map.jinja

```diff
@@ -1,15 +1,111 @@
-{% set collectd = salt['grains.filter_by']({
+{% set os_map = salt['grains.filter_by']({
     'Debian': {
         'pkg': 'collectd-core',
         'service': 'collectd',
         'config': '/etc/collectd/collectd.conf',
         'plugindirconfig': '/etc/collectd/plugins',
-        'javalib': '/usr/lib/collectd/java.so',
+        'TypesDB': ['/usr/share/collectd/types.db'],
     },
     'RedHat': {
         'pkg': 'collectd',
         'service': 'collectd',
         'config': '/etc/collectd.conf',
         'plugindirconfig': '/etc/collectd.d',
+        'TypesDB': ['/usr/share/collectd/types.db'],
     },
 }, merge=salt['pillar.get']('collectd:lookup')) %}
+
+{# Settings dictionary with default values #}
+{% set collectd_settings = {
+    'default_plugins': [
+        'battery',
+        'cpu',
+        'entropy',
+        'load',
+        'memory',
+        'swap',
+        'users'
+    ],
+    'FQDNLookup': 'false',
+    'plugins': {
+        'apache': {
+            'host': salt['grains.get']('host')
+        },
+        'df': {
+            'Device': '',
+            'MountPoint': '',
+            'FSType': '',
+            'IgnoreSelected': 'false',
+            'ReportByDevice': 'false',
+            'ReportReserved': 'false',
+            'ReportInodes': 'false'
+        },
+        'disk': {
+            'matches': []
+        },
+        'enable': false,
+        'ethstat': {
+            'interface': 'eth0'
+        },
+        'interface': {
+            'interface': 'eth0',
+            'IgnoreSelected': 'false'
+        },
+        'java': {
+            'host': 'localhost',
+            'port': '17264',
+            'user': '',
+            'pass': ''
+        },
+        'mysql': {
+            'databases': []
+        },
+        'network': {},
+        'nginx': {
+            'url': 'http://localhost/check_status',
+            'user': '',
+            'password': '',
+            'verifypeer': '',
+            'verifyhost': '',
+            'cacert': ''
+        },
+        'ntpd': {
+            'host': 'localhost',
+            'port': '123',
+            'ReverseLookups': 'false'
+        },
+        'ping': {
+            'hosts': ['google.com', 'yahoo.com'],
+            'hosts_from_grains': {},
+            'interval': '',
+            'timeout': '',
+            'ttl': '',
+            'sourceaddress': '',
+            'device': '',
+            'maxmissed': ''
+        },
+        'postgresql': {
+            'databases': []
+        },
+        'processes': {
+            'Processes': [],
+            'ProcessMatches': []
+        },
+        'statsd': {
+            'host': '"::"',
+            'port': 8125
+        },
+        'syslog': {
+            'loglevel': 'info'
+        },
+        'write_graphite': {
+            'node': salt['grains.get']('fqdn')
+        }
+    }
+} %}
```

Comment:
Just in general, all these defaults should reflect the collectd documented defaults from the config (where there is a default anyway).